packages/devtools/lib/src/ui/plotly.dart

```diff
@@ -0,0 +1,237 @@
+  external String get stackgroup;
+  external List<Transform> get transforms;
+  external String get hoverinfo;
+  external HoverLabel get hoverlabel;
+  external bool get showlegend;
+  external List<int> get width;
+
+  external factory Data({List x, List y, String type, String mode,
+    Marker marker, Line line, String name, List<String> text,
+    String stackgroup, List<Transform> transforms, String hoverinfo,
+    HoverLabel hoverlabel, bool showlegend, List<int> width});
+}
+
+@JS()
+@anonymous
+class Marker {
+  external String get color;
+  external String get symbol;
+  external int get size;
+  external Line get line;
+
+  external factory Marker({String color, String symbol, int size, Line line});
+}
+
+@JS()
+@anonymous
+class Line {
+  external String get color;
+  external String get dash;
+  external int get width;
+
+  external factory Line({String color, String dash, int width});
+}
+
+@JS()
+@anonymous
+class Transform {
+  external List get target;
+  external bool get enabled;
+  external bool get preservegaps;
+  external String get operation;
+  external int get value;
+  external String get type;
+
+  external factory Transform({List target,bool enabled, bool preservegaps, String operation, int value, String type});
+}
+
+@JS()
+@anonymous
+class Layout {
+  external String get title;
+  external bool get showlegend;
+  external AxisLayout get xaxis;
+  external AxisLayout get yaxis;
+  external bool get autosize;
+  external Margin get margin;
+  external String get hovermode;
+  external bool get barmode;
+  external num get bargap;
+  external num get bargroupgap;
+  external String get dragmode;
+
+  external factory Layout({String title, bool showlegend, AxisLayout xaxis,
+    AxisLayout yaxis, bool autosize, Margin margin, String hovermode,
+    String barmode, num bargap, num bargroupgap, String dragmode, HoverLabel hoverlabel});
+}
+
+@JS()
+@anonymous
+class HoverLabel {
+  external String get bgcolor;
+  external String get bordercolor;
+  external Font get font;
+
+  external factory HoverLabel({String bgcolor, String bordercolor, Font font});
+}
+
+@JS()
+@anonymous
+class Font {
+  external String get family;
+  external String get color;
+  external int get size;
+
+  external factory Font({String family, String color, int size});
+}
+
+@JS()
+@anonymous
+class AxisLayout {
+  external String get tickformat;
+  external String get ticks;
+  external String get title;
+  external bool get fixedrange;
+  external String get type;
+  external String get autorange;
+  external List<num> get range;
+  external RangeSlider get rangeslider;
+  external RangeSelector get rangeselector;
+  external bool get showgrid;
+  external bool get showticklabels;
+
+  external factory AxisLayout({String tickformat, String ticks, String title,
+    bool fixedrange, String type, String autorange, List<num> range,
+    RangeSlider rangeslider, RangeSelector rangeselector, bool showgrid, bool showticklabels});
```

Comment:
nit: add a trailing comma here for nicer dartfmt output.

Reply:
Done.